IdentityCore/src/controllers/broker/MSIDSSOExtensionInteractiveTokenRequestController.m

```diff
@@ -52,18 +53,53 @@ - (void)acquireToken:(MSIDRequestCompletionBlock)completionBlock
     {
         MSID_LOG_WITH_CTX(MSIDLogLevelInfo, self.requestParameters, @"Interactive broker extension flow finished. Result %@, error: %ld error domain: %@", _PII_NULLIFY(result), (long)error.code, error.domain);
         
+        if ([self shouldFallback:error])
+        {
+            MSID_LOG_WITH_CTX(MSIDLogLevelInfo, self.requestParameters, @"Falling back to local controller.");
+            
+            [self.fallbackController acquireToken:completionBlock];
+        }
+        
         completionBlock(result, error);
     };
     
-    __auto_type request = [self.tokenRequestProvider interactiveSSOExtensionTokenRequestWithParameters:self.interactiveRequestParameters];
+    __auto_type request = [self.tokenRequestProvider interactiveSSOExtensionTokenRequestWithParameters:self.interactiveRequestParamaters];
 
     [self acquireTokenWithRequest:request completionBlock:completionBlockWrapper];
 }
 
 + (BOOL)canPerformRequest
 {
-    // TODO: implement.
-    return YES;
+    return [[ASAuthorizationSingleSignOnProvider msidSharedProvider] canPerformAuthorization];
+}
+
+#pragma mark - Private
+
+- (BOOL)shouldFallback:(NSError *)error
+{
+    MSID_LOG_WITH_CTX(MSIDLogLevelInfo, self.requestParameters, @"Looking if we should fallback to fallbackController, error: %ld error domain: %@.", (long)error.code, error.domain);
+    
+    if (!self.fallbackController)
+    {
+        MSID_LOG_WITH_CTX(MSIDLogLevelInfo, self.requestParameters, @"fallbackController is nil, shouldFallback: NO");
+        return NO;
+    }
+    
+    if (![error.domain isEqualToString:ASAuthorizationErrorDomain]) return NO;
+    
+    // TODO: verify this logic.
+    BOOL shouldFallback = NO;
+    switch (error.code)
+    {
+        case ASAuthorizationErrorNotHandled:
+        case ASAuthorizationErrorUnknown:
+        case ASAuthorizationErrorFailed:
+            shouldFallback = YES;
+    }
+    
+    MSID_LOG_WITH_CTX(MSIDLogLevelInfo, self.requestParameters, @"shouldFallback: %@", shouldFallback ? @"YES" : @"NO");
```

Comment:
nit: provide more verbose log here, e.g. SSO extension controller should fallback : %@

Reply:
done.